tests/test_salesforce_switch_rep_method_ft_incrmntl.py

```diff
@@ -0,0 +1,96 @@
+
+        # Run in check mode
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # Select only the expected streams tables
+        expected_streams = self.expected_sync_streams()
+        catalog_entries = [ce for ce in found_catalogs if ce['tap_stream_id'] in expected_streams]
+        self.select_all_streams_and_fields(conn_id, catalog_entries)
+        streams_replication_methods = {stream: self.FULL_TABLE
+                                       for stream in expected_streams}
+        self.set_replication_methods(conn_id, catalog_entries, streams_replication_methods)
+
+        # Run a sync job using orchestrator
+        fulltbl_sync_record_count = self.run_and_verify_sync_mode(conn_id)
+        fulltbl_sync_records = runner.get_records_from_target_output()
+
+        fulltbl_sync_bookmarks = menagerie.get_state(conn_id)
+
+        #Switch the replication method from full table to Incremental
+        streams_replication_methods = {stream: self.INCREMENTAL
+                                       for stream in expected_streams}
+        self.set_replication_methods(conn_id, catalog_entries, streams_replication_methods)
+
+        # SYNC 2
+        incrmntl_sync_record_count = self.run_and_verify_sync_mode(conn_id)
+        incrmntl_sync_records = runner.get_records_from_target_output()
+        incrmntl_sync_bookmarks = menagerie.get_state(conn_id)
+
+        # Test by stream
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+                # record counts
+                fulltbl_sync_count = fulltbl_sync_record_count.get(stream, 0)
+                incrmntl_sync_count = incrmntl_sync_record_count.get(stream, 0)
+                replication_key = list(replication_keys[stream])[0]
+
+                # Verify at least 1 record was replicated in the Incrmental sync
+                self.assertGreater(incrmntl_sync_count, 0,
+                                   msg="We are not fully testing bookmarking for {}".format(stream))
+                # data from record messages
+                primary_key = list(primary_keys[stream])[0]
+                fulltbl_sync_messages = [record['data'] for record in
+                                       fulltbl_sync_records.get(stream).get('messages')
+                                       if record.get('action') == 'upsert']
+                fulltbl_primary_keys = {message[primary_key] for message in fulltbl_sync_messages}
+                incrmntl_sync_messages = [record['data'] for record in
+                                        incrmntl_sync_records.get(stream).get('messages')
+                                        if record.get('action') == 'upsert']
+                incrmntl_primary_keys = {message[primary_key] for message in incrmntl_sync_messages}
+
+                #Verify all records are synced in the second sync
+                self.assertTrue(fulltbl_primary_keys.issubset(incrmntl_primary_keys))
```

Comment:
In this scenario, a few records are updated on each sync, specifically the replication key value, thus when we filter the incremental sync, we are actually removing records that were generated previously but were recently altered.
As a result, the equal primary key sets assumption fails. 
After talking with Harrison, we concluded that it should be fine if the full_table sync primary keys are a subset of the Incremental sync primary keys in case any new entries are added after the full_table sync.

Reply:
Correct. This assures that the incremental sync has all the records from the original full table sync, ignoring extra records.  

I'm not sure what the answer was for salesforce about full table adhering to the start date or ignoring the start date.  This wouldn't work if the full table sync ignored the start date and went before it.  This could be okay with an early enough start date, but when we put this in tap tester, we should generalize this to filter out records before the start date in the full sync records.

Reply:
Full table doesn't adhere to start date. Will modify to filter the full table sync records according to the start date.